### PR TITLE
feat: add single select v1.2 params

### PR DESCRIPTION
**Pull request by allisonc07**
Make singleSelect accept either an object or a boolean to generate the auth url!

### CODE REVIEW

File: test/unit/lib/auth-client.js [hunk 1]
```diff
@@ -251,6 +251,117 @@ test('getAuthUrl - test mode false', function(t) {
...
+  expected += '&approval_prompt=force';
+  expected += '&scope=read_odometer%20read_vehicle_info';
+  expected += '&state=fakestate';
+  expected += '&single_select=true';
+  expected += '&mode=live';
+
+  t.is(actual, expected);
+});
+
+test('getAuthUrl - single select vin', function(t) {
```
Comment (adolfoportilla) [inline test/unit/lib/auth-client.js]: Maybe add one more tests that includes wrong keys passed in the object?

Comment (allisonc07) [reply test/unit/lib/auth-client.js]: got it!

File: lib/auth-client.js [hunk 2]
```diff
@@ -158,8 +162,26 @@ AuthClient.prototype.getAuthUrl = function(options) {
   }
 
   if (options.singleSelect !== undefined) {
-    // eslint-disable-next-line camelcase
-    parameters.single_select = options.singleSelect === true;
+    if (typeof options.singleSelect === 'object'
+      && options.singleSelect !== null) {
```
Comment (adolfoportilla) [inline lib/auth-client.js]: options.singleSelect = {}
options.singleSelect !== null (this gives true). I don't think it is possible for an object to be null

File: test/unit/lib/auth-client.js [hunk 3]
```diff
@@ -332,6 +365,58 @@ test('getAuthUrl - single select junk', function(t) {
   t.is(actual, expected);
 });
 
+test('getAuthUrl - single select with junk keys in object passed in', function(t) {
```
Comment (adolfoportilla) [inline test/unit/lib/auth-client.js]: Thoughts on how singleSelect should behave? (we should test for all of this porbably, I think we can do it in 2 tests only
```
singleSelect: true                    -> single_select=true
singleSelect: false/'hello'/0/1/3     -> single_select=false
singleSelect: undefined or not passed -> single_select is not added

singleSelect: {}, {potato}            -> single_select=false
singleSelect: {vin}, {vin, potato}    -> single_select=true&single_select_vin=vin
```

Comment (allisonc07) [reply test/unit/lib/auth-client.js]: this is the behavior that I have imagined! the only ones that aren't covered currently are `{}` and `{vin, potato}`. Can add these cases as well!

File: lib/auth-client.js [hunk 4]
```diff
@@ -157,9 +159,26 @@ AuthClient.prototype.getAuthUrl = function(options) {
     parameters.state = options.state;
   }
 
-  if (options.singleSelect !== undefined) {
-    // eslint-disable-next-line camelcase
-    parameters.single_select = options.singleSelect === true;
+  if (options.singleSelect !== undefined && options.singleSelect !== null) {
+    if (typeof options.singleSelect === 'object') {
+      const availableParams = ['vin'];
+      let singleSelectParamAdded = false;
```
Comment (adolfoportilla) [inline lib/auth-client.js]: we can do the same as python to clean it up a little bit? initialize the single_select to false and then change it to true?

Comment (allisonc07) [reply lib/auth-client.js]: yup! fixed

**Pull request merged**